example/RotatedSC_d3.jl

```diff
@@ -0,0 +1,254 @@
+            H(b[1])
+            CNOT(b[1], b[2])
+        else
+            res = -100
+        end
+    end
+
+    res
+end
+
+@qprog surface_code_decoder (d) begin
+
+    xq = [1, 2, 4, 8]
+    zq = [1, 3, 4, 5]
+
+    s_x = [surface_code_x_m(d, j) for j in xq]
+    s_z = [surface_code_z_m(d, j) for j in zq]
+
+    r_x = mwpm(d, s_x, "X")
+    r_z = mwpm(d, s_z, "Z")
+
+    print("rx=$(r_x)")
+    print("rz=$(r_z)")
+
+
+    for j in 1:d*d
+        sZ(j, r_x[j])
+        sX(j, r_z[j])
+    end
+
+    # a strange bug
+    e = reduce(&, r_z[1:(d-1)÷2])
+
+    sX(1, e)
+
+end
+
+
+
+function check_surface_code_decoder(d::Integer)
+
+    d = 3
+
+    @info "Initailization Stage"
+    t0 = time()
+    begin
+        num_qubits = d*d
+
+	    stabilizer = fill(false, num_qubits, 2*num_qubits)
+
+        xq = [1, 2, 4, 8]
+
+        for r in 1:4
+            for x_adj in _xadj(3,xq[r])
+                stabilizer[r, x_adj] = true
+            end
+        end
+
+        zq = [1, 3, 4, 5]
+
+        for r in 1:4
+            for z_adj in _zadj(3,zq[r])
+                stabilizer[4+r, num_qubits+z_adj] = true
+            end
+        end
+
+        stabilizer[9,2] = true
+        stabilizer[9,5] = true
+        stabilizer[9,8] = true
+        
+        println("Encoded stabilizer : $(stabilizer)")
+
+
+
+	    phases = Vector{Z3.ExprAllocated}(undef, num_qubits)
+	    lx = _bv_const(ctx, "lx")
+	    lz = _bv_const(ctx, "lz")
+
+	    for i in 1:num_qubits-1
+	    	phases[i] = _bv_val(ctx, 0)
+	    end
+
+        phases[9] = lx
+
+        ρ01 = from_stabilizer(num_qubits, stabilizer, phases, ctx)
+        ρ1 = copy(ρ01)
+
+
+        σ = CState([(:d, d),
+            (:surface_code_decoder, surface_code_decoder),
+            (:surface_code_z_m, surface_code_z_m),
+            (:surface_code_x_m, surface_code_x_m),
+            (:_xadj, _xadj),
+            (:_zadj, _zadj),
+            (:ctx, ctx),
+            (:mwpm, mwpm)
+        ])
+
+        num_x_errors = (d-1)÷2
+        x_errors = inject_errors(ρ1, "X")
+        ϕ_x1 = _sum(ctx, x_errors, num_qubits) == bv_val(ctx, num_x_errors, _len2(num_qubits)+1)
+
+	  
+        decoder = surface_code_decoder(d)
+        cfg1 = SymConfig(decoder, σ, ρ1)
+        # cfg2 = SymConfig(surface_code_decoder(d), σ, ρ2)
+    end
+
+    @info "Symbolic Execution Stage"
+    t1 = time()
+    begin
+        cfgs1 = QuantSymEx(cfg1)
+        # cfgs2 = QuantSymEx(cfg2)
+    end
+
+    @info "SMT Solver Stage"
+    t2 = time()
+
+    begin
+        res = true
+        # for cfg in cfgs1
+        if !check_state_equivalence(
+            ρ01, ρ01, (ϕ_x1 #=& ϕ_z1=#, cfg1.ϕ[1], cfg1.ϕ[2]),
+            # cfg.ρ, ρ01, (ϕ_x1 #=& ϕ_z1=#, cfg.ϕ[1], cfg.ϕ[2]),
+            `bitwuzla --smt-comp-mode true -rwl 0 -S kissat`
+            #`bitwuzla --smt-comp-mode true -S kissat`
+            #`bitwuzla --smt-comp-mode true -rwl 0`
+            )
+            res = false
+            # break
+        end
+        # end
```

Comment:
If I do the below change to be consistent across the board, I get the following error "The Stabilizer does not match, the program is wrong even without error insertion" irrespective of the fact that I am using a correct/incoreect decoder.
```suggestion
        for cfg in cfgs1
            if !check_state_equivalence(
                  # ρ01, ρ01, (ϕ_x1 #=& ϕ_z1=#, cfg1.ϕ[1], cfg1.ϕ[2]),
                   cfg.ρ, ρ01, (ϕ_x1 #=& ϕ_z1=#, cfg.ϕ[1], cfg.ϕ[2]),
                   `bitwuzla --smt-comp-mode true -rwl 0 -S kissat`
                    #`bitwuzla --smt-comp-mode true -S kissat`
                     #`bitwuzla --smt-comp-mode true -rwl 0`
                    )
                  res = false
                 break
                 end
        end
```